[Docs] Update "Customizing Repositories" (#10290)
Sylius/Sylius: docs/customization/repository.rst

    $ php bin/console debug:container sylius.repository.product

As a result you will get the ``Sylius\Bundle\CoreBundle\Doctrine\ORM\ProductRepository`` - this is the class that you need to be extending.
To make your class more reusable, you should create new interface ``App/Repository/ProductRepositoryInterface.php`` which one will extend ``Sylius\Component\Core\Repository\ProductRepositoryInterface``

.. code-block:: php
    <?php
    declare(strict_types=1);
    namespace App\Repository;
    use Sylius\Component\Core\Repository\ProductRepositoryInterface as BaseProductRepositoryInterface;

>     interface ProductRepositoryInterface extends BaseProductRepositoryInterface

Reviewer: Missing blank lines

    $ php bin/console debug:container sylius.repository.product

As a result you will get the ``Sylius\Bundle\CoreBundle\Doctrine\ORM\ProductRepository`` - this is the class that you need to be extending.

> To make your class more reusable, you should create new interface ``App/Repository/ProductRepositoryInterface.php`` which will extend ``Sylius\Component\Core\Repository\ProductRepositoryInterface``

Reviewer 2: ```suggestion
To make your class more reusable, you should create a new interface ``src/Repository/ProductRepositoryInterface.php`` 
which will extend the``Sylius\Component\Core\Repository\ProductRepositoryInterface``.
```

.. code-block:: php

    <?php

    declare(strict_types=1);

    namespace App\Repository;

    use Sylius\Component\Core\Repository\ProductRepositoryInterface as BaseProductRepositoryInterface;

    interface ProductRepositoryInterface extends BaseProductRepositoryInterface
    {

>         public function findAllByOnHand(int $limit): array;

Reviewer: This method name is not compatible with method name in `ProductRepository`

We are sorting the results by the count of how many products are still available on hand, which is saved on the ``onHand`` field on the specific ``variant`` of each product.
Then we are limiting the query to 8 by default, to get only 8 products that are low in stock.

**2.** In order to use your repository you need to configure it in the ``config/packages/_sylius.yaml``.
As you can see in ``_sylius.yaml`` you already have basic configuration, now you just need add your repository and override resourceRepository

.. code-block:: yaml

    sylius_product:
        resources:
            product:
                classes:
                    model: App\Entity\Product\Product
                    repository: App\Repository\ProductRepository # you need add your repository to the config file
                translation:
                    classes:
                        model: App\Entity\Product\ProductTranslation

**3.** After configuring the ``sylius.repository.product`` service has your ``findByOnHand()`` method available.

> You can now use your method in anywhere when you are operating on the Product repository. For example you can configure new ``route``

Reviewer 2: ```suggestion
You can now use your method in anywhere when you are operating on the Product repository. 
For example, you can configure a new route:
```

.. code-block:: yaml

    app_admin_custom_product_index:
        path: /admin/custom-products/
        methods: [GET]
        defaults:
            _controller: sylius.controller.product:indexAction
            _sylius:
                template: '@SyliusShop/Product/_horizontalList.html.twig'
                repository:
                    method: findAllByOnHand
                    arguments: [2]
                criteria: false
            paginate: false
            limit: 100

> And use it in override mainpage:

Reviewer 2: This part below is not needed :D

We are sorting the results by the count of how many products are still available on hand, which is saved on the ``onHand`` field on the specific ``variant`` of each product.
Then we are limiting the query to 8 by default, to get only 8 products that are low in stock.

**2.** In order to use your repository you need to configure it in the ``config/packages/_sylius.yaml``.

> As You see in ``_Sylius.yaml`` You already have basic configuration, now You just need add Yours repository and override resourceRepository

Reviewer 3: I'd also avoid starting words such as `you, your etc.` with a capital letter.

> As You see in ``_Sylius.yaml`` You already have basic configuration, now You just need add Yours repository and override resourceRepository

Reviewer 3: `As You see` -> `As you can see`

> As You see in ``_Sylius.yaml`` You already have basic configuration, now You just need add Yours repository and override resourceRepository

Reviewer 3: `Yours repository` -> `Your repository`

.. code-block:: yaml

    sylius_product:
        resources:
            product:
                classes:
                    model: App\Entity\Product\Product

>                     repository: App\Repository\ProductRepository # you need add yours repository to config file

Reviewer 3: `Yours repository` -> `Your repository`

>                     repository: App\Repository\ProductRepository # you need add yours repository to config file

Reviewer 3: `to the config file`

                translation:
                    classes:
                        model: App\Entity\Product\ProductTranslation

**3.** After configuring the ``sylius.repository.product`` service has your ``findByOnHand()`` method available.

> You can form now on use your method in any **Controller**, or files which use ``sylius.repository.product``.

Reviewer 3: Shouldn't it be `You can now use your method in any **Controller**, or files which use ``sylius.repository.product``.` ? Since the sentence above is just a modified version of a previously existing one, I'd summon @CoderMaggie here :D

Reviewer 2: What kind of files are these? :D
Let's make it:
```suggestion
You can now use your method in anywhere when you are operating on the Product repository.
```
what do you think?

We are sorting the results by the count of how many products are still available on hand, which is saved on the ``onHand`` field on the specific ``variant`` of each product.
Then we are limiting the query to 8 by default, to get only 8 products that are low in stock.

**2.** In order to use your repository you need to configure it in the ``config/packages/_sylius.yaml``.
As you can see in the ``_sylius.yaml`` you already have a basic configuration, now you just need to add your repository and override resourceRepository

.. code-block:: yaml

    sylius_product:
        resources:
            product:
                classes:

>                     model: App\Entity\Product\Product

Reviewer 2: why?

                    repository: App\Repository\ProductRepository

>                 translation:

Reviewer 2: why?

                    classes:

>                         model: App\Entity\Product\ProductTranslation

Reviewer: I guess that here you should add only a `repository: App\Repository\ProductRepository`

**3.** After configuring the ``sylius.repository.product`` service has your ``findByOnHand()`` method available.
You can now use your method in anywhere when you are operating on the Product repository.
For example you can configure new route:

.. code-block:: yaml

>     app_admin_custom_product_index:

Reviewer: `admin` 😱

We are sorting the results by the count of how many products are still available on hand, which is saved on the ``onHand`` field on the specific ``variant`` of each product.
Then we are limiting the query to 8 by default, to get only 8 products that are low in stock.

**2.** In order to use your repository you need to configure it in the ``config/packages/_sylius.yaml``.

> As you can see in the ``_sylius.yaml`` you already have a basic configuration, now you just need to add your repository and override resourceRepository

Reviewer 4: As you can see in the ``_sylius.yaml`` you already have a basic configuration. Now you need to configure your custom repository as a resource's repository:

wdyt?

.. code-block:: yaml

    sylius_product:
        resources:
            product:
                classes:
                ...

>                     repository: App\Repository\ProductRepository

Reviewer 4: Shouldn't it be placed in `App\Doctrine\ORM\ProductRepository`? The interface in `Repository` namespace is good for me, as it's not coupled with ORM in any way.

                ...

**3.** After configuring the ``sylius.repository.product`` service has your ``findByOnHand()`` method available.

> You can now use your method in anywhere when you are operating on the Product repository.

Reviewer 4: `in anywhere when` -> `anywhere`? cc @CoderMaggie

> For example you can configure new route:

Reviewer 4: For example, you can configure **a** new route:

.. code-block:: yaml

    app_shop_partial_product_index_by_on_hand:

>     path: /partial/products/by-on-hand

Reviewer 4: Indent missing
